pkg/controllers/cluster/cluster_controller.go

```diff
@@ -420,6 +420,10 @@ func (c *Controller) tryUpdateClusterHealth(ctx context.Context, cluster *cluste
 		}
 	}
 
+	if cluster.Spec.SyncMode == clusterv1alpha1.Push {
+		return observedReadyCondition, currentReadyCondition, nil
+	}
+
```

Comment:
Why not check it earlier? At least we can do it before `Step 3`

Can we do it at the head of this function(tryUpdateClusterHealth)?

Reply:
But we still need to maintain `clusterHealth.readyTransitionTimestamp` for [processTaintBaseEviction](https://github.com/karmada-io/karmada/blob/v1.2.0/pkg/controllers/cluster/cluster_controller.go#L492)

Reply:
oh.. right